### PR TITLE
feat: add new git connection type for connections

### LINKED ISSUE #1449: Add a new "git" connection type
Ref: https://github.com/flanksource/duty/pull/303

### PR DESCRIPTION
**Pull request by mainawycliffe**
Closes #1449

### CODE REVIEW

File: src/components/Connections/connectionTypes.tsx [hunk 1]
```diff
@@ -1807,6 +1807,55 @@ export const connectionTypes: ConnectionType[] = [
...
+    value: ConnectionValueType.Git,
+    fields: [
+      {
+        label: "Name",
+        key: "name",
+        type: fieldTypes.input,
+        required: true
+      },
+      {
+        label: "Certificate",
```
Comment (moshloop) [inline src/components/Connections/connectionTypes.tsx:1823]: We need a username and password as well and rename the certificate label to SSH Key
+ reorder name, url, user, pass, certificate, ref

File: src/components/Connections/connectionTypes.tsx [hunk 2]
```diff
@@ -1807,6 +1807,69 @@ export const connectionTypes: ConnectionType[] = [
...
+        required: true
+      },
+      {
+        label: "URL",
+        key: "url",
+        type: fieldTypes.input,
+        required: true
+      },
+      {
+        label: "Username",
```
Comment (moshloop) [inline src/components/Connections/connectionTypes.tsx:1829]: Its either Username or Certificate, not both and I don't think any of them are required.

**Comment by moshloop** [comment]
@mainawycliffe I can't see to save the form ?

**Comment by mainawycliffe** [comment]
@moshloop  Let me investigate

**Comment by moshloop** [comment]
@mainawycliffe - The password doesn't seem to save - also if you specify a user/pass and then save and  edit it shows you an empty ssh key, it should choose the value that is not null 
- i  should also default to user/pass not ssh

**Comment by moshloop** [comment]
Can we also update the user/pass style to the same as on the http form (extracting and resuing components as needed)

**Comment by moshloop** [comment]
@mainawycliffe  We now have a Username None/Basic that expands to username and password under username?  It should just be Auth: None | Basic | SSH,

**Comment by moshloop** [comment]
@mainawycliffe  We also need to add a namespace (defaulting to `default`) to all the connection forms and the table

**Pull request merged**